plugins/basic/sg_client.py

```diff
@@ -34,77 +19,52 @@ def log(msg):
+    return UnityEngine
 
 def GetUnityEditor():
-    return _service.UnityEditor
-
-class ShotgunClientService(unity_client.UnityClientService):
+    import UnityEditor
+    return UnityEditor
+    
+#### FUNCTIONS PREVIOUSLY IN CLIENT
+def test_invoke_post_init_hook():
     """
-    The RPyC service class. Exposes all the services that are required for the 
-    Shotgun integration
+    Invokes the post init hook
     """
-    def exposed_client_name(self):
-        return 'com.unity.integrations.shotgun'
-    
-    def exposed_on_server_shutdown(self, invite_retry):
-        global _connection
-        _connection.close()
-        _connection = None
+    try:
+        import sgtk
+        sgtk.platform.current_engine().execute_hook_method('post_init_hook', 'on_post_init')
+    except:
+        import traceback
+        log('Got exception while invoking the post init hook:')
+        log('Exception stack trace:\n\n{}'.format(traceback.format_exc()))
 
-        if invite_retry:
-            connect_to_unity(wait_for_server=True)
-        else:
-            super(ShotgunClientService, self).exposed_on_server_shutdown(invite_retry)
+def test_tk_unity_version():
+    """
+    Returns a string representing the version of tk-unity, e.g. "v0.9"
+    """
+    import sgtk
+    version = sgtk.platform.current_engine().version
+    return version
     
-    @scheduling.exec_on_main_thread
-    def exposed_execute_menu_item(self,menu_item):
-        """
-        Calls the menu item callback
-        """
-        try:
-            import sgtk
-            sgtk.platform.current_engine().call_menu_item_callback(menu_item)
-        except:
-            import traceback
-            log('Got exception while executing menu item "{}"'.format(menu_item))
-            log('Exception stack trace:\n\n{}'.format(traceback.format_exc()))
-
-    @scheduling.exec_on_main_thread
-    def exposed_invoke_post_init_hook(self):
-        """
-        Invokes the post init hook
-        """
-        try:
-            import sgtk
-            sgtk.platform.current_engine().execute_hook_method('post_init_hook', 'on_post_init')
-        except:
-            import traceback
-            log('Got exception while invoking the post init hook:')
-            log('Exception stack trace:\n\n{}'.format(traceback.format_exc()))
-
-    def exposed_tk_unity_version(self):
-        """
-        Returns a string representing the version of tk-unity, e.g. "v0.9"
-        """
+def test_execute_menu_item(menu_item):
+    """
+    Calls the menu item callback
+    """
+    try:
         import sgtk
-        return sgtk.platform.current_engine().version
+        sgtk.platform.current_engine().call_menu_item_callback(menu_item)
+    except:
+        import traceback
+        log('Got exception while executing menu item "{}"'.format(menu_item))
+        log('Exception stack trace:\n\n{}'.format(traceback.format_exc()))
+#### --------------------    
 
-@scheduling.exec_on_main_thread
```

Comment:
I think this still needs to be on the main thread.

Reply:
Although maybe it is already on the main thread every time it gets called so we don't need to do anything special.

Reply:
It is. Unless `[InitializeOnLoad]` stops being on the main thread.